webextension/js/popup.js

```diff
@@ -481,22 +482,28 @@ Logic.registerPanel(P_CONTAINER_EDIT, {
       Logic.showPreviousPanel();
     });
 
-    document.querySelector("#edit-container-ok-link").addEventListener("click", () => {
-      const identity = Logic.currentIdentity();
-      const formValues = new FormData(document.getElementById("edit-container-panel-form"));
-      browser.runtime.sendMessage({
-        method: identity.userContextId ? "updateIdentity" : "createIdentity",
-        userContextId: identity.userContextId || 0,
-        name: document.getElementById("edit-container-panel-name-input").value || Logic.generateIdentityName(),
-        icon: formValues.get("container-icon") || DEFAULT_ICON,
-        color: formValues.get("container-color") || DEFAULT_COLOR,
-      }).then(() => {
-        return Logic.refreshIdentities();
-      }).then(() => {
-        Logic.showPreviousPanel();
-      }).catch(() => {
-        Logic.showPanel(P_CONTAINERS_LIST);
-      });
+    this._editForm = document.getElementById("edit-container-panel-form");
+    const editLink = document.querySelector("#edit-container-ok-link");
+    editLink.addEventListener("click", this._submitForm);
+    editLink.addEventListener("submit", this._submitForm);
+    this._editForm.addEventListener("submit", this._submitForm);
+  },
+
+  _submitForm() {
+    const identity = Logic.currentIdentity();
+    const formValues = new FormData(this._editForm);
```

Comment:
This seems to have caused a bug (in Firefox 52?) where editing a container always sets the color and icon to the default blue and circle. When the code gets to this line of execution, `this._editForm` is `undefined` 😢 Not sure why ...